17-2-release-notes-draft (#22127)
opf/openproject: docs/release-notes/17-2-0/README.md

## Important feature changes

Take a look at our release video showing the most important features introduced in OpenProject 17.2.0:

![Release video of OpenProject 17.2]( UPDATE LINK) 

### MCP Server (Enterprise add-on)

[feature: mcp_server ]

> OpenProject 17.2 introduces the **MCP Server**, a new Enterprise add-on that lays the foundation for robust integrations between OpenProject and external intelligent agents, automation tools, or systems that use the Model Context Protocol (MCP). This server uses OpenProject’s APIv3 resources as MCP-compatible endpoints and enables secure, authenticated access for clients such as large language models or other MCP clients, opening the door to richer contextual interactions with your project data. 

Reviewer: Detail: We do not really rely on APIv3 to offer the MCP server, however it's correct that we reuse the APIv3 format for MCP. Small wording suggestion to better reflect that:

```suggestion
OpenProject 17.2 introduces the **MCP Server**, a new Enterprise add-on that lays the foundation for robust integrations between OpenProject and external intelligent agents, automation tools, or systems that use the Model Context Protocol (MCP). This server exposes OpenProject’s APIv3 resources as MCP-compatible endpoints and enables secure, authenticated access for clients such as large language models or other MCP clients, opening the door to richer contextual interactions with your project data. 
```

Additionally I want to give some feedback on the first sentence:

> that lays the foundation for robust integrations between OpenProject and external intelligent agents, automation tools, or systems that use the Model Context Protocol (MCP).

Semantically: I don't understand the **or** enumeration here. The MCP server is _only_ intended for "systems that use the Model Context Protocol". Automation tools and agents need to implement/have access to an MCP client to make use of this. So maybe this could be phrased in a way of "systems that use the Model Context Protocol, such as ..."

"intelligent agent": Do we know what we mean with this term? I needed to look it up and wasn't sure if it's made up, but apparently it exists: https://en.wikipedia.org/wiki/Intelligent_agent
However, this existing term seems to be very generic/broad, so I am not sure if this is what we were aiming for here.
Do we want to indicate that "your favorite large language model" can be hooked up to this? I think in this case we should use the term "LLM" or "large language model". An example (combined with the proposals above:

```suggestion
OpenProject 17.2 introduces the **MCP Server**, a new Enterprise add-on that lays the foundation for robust integrations between OpenProject and systems that use the Model Context Protocol (MCP), such as large language models or automation tools. This server exposes OpenProject’s APIv3 resources as MCP-compatible endpoints and enables secure, authenticated access for clients such as large language models or other MCP clients, opening the door to richer contextual interactions with your project data. 
```

Technicality to my suggestion above: The LLMs usually run on a platform where they have access to an MCP client, but don't implement the MCP client as part of the model. I think the proposal above reflects this enough by saying "**use** the Model Context Protocol", but in case you want to reformulate my suggestion, I'd _not_ write something that implies an LLM _is_ an MCP client.

> Included in this release are administrative UI support for configuring the MCP Server, infrastructure and metadata endpoints, and integration of MCP authentication with OpenProject’s OAuth2 and API key mechanisms. An initial set of MCP tools and resources is provided to surface key entities (projects, work packages, users, etc.), and response formats can be adjusted based on your preferences. With session-cookie and bearer-token support, the MCP Server acts as a secure bridge between your OpenProject instance and external systems that operate via MCP. 

Reviewer: I think something that we can brag about more openly, is that as a consequence on how we built this, it also integrates cleanly with external authentication providers (think of Keycloak):

```suggestion
Included in this release are administrative UI support for configuring the MCP Server, infrastructure and metadata endpoints, and integration of MCP authentication with OpenProject’s OAuth2 and API key mechanisms, including external OpenID Connect providers. An initial set of MCP tools and resources is provided to surface key entities (projects, work packages, users, etc.), and response formats can be adjusted based on your preferences. With session-cookie and bearer-token support, the MCP Server acts as a secure bridge between your OpenProject instance and external systems that operate via MCP. 
```

> and response formats can be adjusted based on your preferences

I believe that this sentence was born out of https://community.openproject.org/wp/71978, which I consider to be a pretty niche (as in _small_, not "nice") setting. We built this after we observed for _one_ MCP client that it passed too much data on to the LLM and thus decreased model performance. Other MCP clients didn't seem to make the same mistake and thus don't require any fiddling with this setting at all.

I think it's good that we included the setting, because maybe someone will find it useful, but it's also a good candidate for a feature to be removed, once we see that it's not really needed. I'd try to not advertise it too strongly. My suggestion would be to drop the quoted half-sentence

```suggestion
Included in this release are administrative UI support for configuring the MCP Server, infrastructure and metadata endpoints, and integration of MCP authentication with OpenProject’s OAuth2 and API key mechanisms. An initial set of MCP tools and resources is provided to surface key entities (projects, work packages, users, etc.). With session-cookie and bearer-token support, the MCP Server acts as a secure bridge between your OpenProject instance and external systems that operate via MCP. 
```

> With session-cookie and bearer-token support, the MCP Server acts as a secure bridge between your OpenProject instance and external systems that operate via MCP. 

Again, session-cookie support was something that we added late into the process because it _might_ be valuable in niche use-cases, but it's nothing that deserves spotlight IMO.

I think the Bearer token and OAuth-support do deserve spotlight. For security on the one side (even though I have to admit that I find this a weird one to point out too strongly, because no one would ever admit that they built something insecurely), but also for compatibility/interoperability on the other side. As I indicated above, support for OAuth (in the way we did it), means that we support external IDPs out of the box.

![Model context protocol settings in OpenProject administration](openproject_system_guide_new_mcp.png)

See the [**MCP Server documentation**](../../system-admin-guide/integrations/mcp-server/) for setup and examples.  

### Reusable meeting templates (Enterprise add-on)

[feature: meeting_templates ]

Preparing meetings often involves recreating the same agenda structure again and again. With OpenProject 17.2, administrators can now define reusable meeting templates that provide a predefined agenda layout for their teams. These templates help ensure that important discussion points are consistently covered and that project meetings follow a clear, standardized structure.

When creating a new one-time meeting, users can choose from the available templates to automatically populate the agenda with the predefined sections and items. This saves time during setup and promotes alignment across teams. 

![Meetings templates listed in an OpenProject project](openproject_release_notes_17-2-0_meetings_template.png)

For more details, please refer to the [Meetings documentation](../../user-guide/meetings/one-time-meetings/).

> ###  Increased security for external links (Enterprise add-on)

Reviewer: Red flag when reading this:

> Increased security [is an] Enterprise add-on

I think it's a fair limitation for the feature in question, but it could be misread as "if you really want to have a secure product, you have to pay".

---

Maybe Oliver or Klaus can refresh my memory on _how_ the login actually improves the security here, but I think the intent was to make sure that users are aware that they receive an email generated by OpenProject that directs them to an external destination. The login requirement probably makes spamming links less effective, because only logged in users will see them, but not search engines / anonymous users? (I am not entirely sure about the attack vector here)

Author: @klaustopher what do you think?
I am happy to change the header for this

Reviewer 2: It is not so much a question about the attack vector directly but about the value of a server that sends out emails. Without the feature, there is value in using the server to send out mails to a lot of people (that don't have to be users). With the feature, the value decreases as only members can be targeted. 

For phishing and spamming that can make a lot of difference. With the feature, spammers might be less encouraged to use an OpenProject instance for this purpose. Of course, with the feature only being available on Enterprise, somebody could still set up their own instance to use it for the purpose. But that is quite a lot of work for that goal. So it protects instances not controlled by a spammer.

Reviewer: That explanation makes a lot of sense. So it's about unauthorized use of email resources.

Though it's still slightly confusing to me, because it would make us and our SaaS offering a target audience for the feature. E.g. it's a legitimate risk that someone creates a trial instance and does what you just described.

It sounds less likely that this would happen at paid enterprise instance, because an untrusted entity would need to be granted access to that enterprise instance and then abuse it for this purpose. But the risk is mostly on the side of the operator of the infrastructure (in case of SaaS: us), because it's their email sender reputation going downhill due to this.

> somebody could still set up their own instance to use it for the purpose

That's usually not the goal for a spammer. The goal is usually to avoid own infrastructure at all, because the mails are spam and will be marked as such eventually.

---

I am still wondering how we should properly formulate this piece of the release notes. I usually have a problem with vague statements like "Increased security for external links" if there is no explanation in what exactly can be insecure about an external link and how that's mitigated.

As we learned here, this feature is less about making the external links more secure, but rather about making it harder to use an OP installation for spamming/phishing purposes against the will of its operator. I think this is what should be reflected here.

Reviewer: Specifically, I think that this part of our explanation is wrong/misleading:

> This helps reduce the risk of users being exposed to phishing, social engineering, or other malicious content by ensuring links are only accessible to signed-in members of your organization.

Users with an account are as likely to be exposed to all of the malicious content mentioned above. What we mitigate against is using an instance where you don't have admin privileges, but enough privileges to invite users for sending such emails to unregistered users.

Administrators can now require users to be logged in before following external links. When this setting is enabled, anyone who is not authenticated will be redirected to the login page before being allowed to continue to the external destination. This helps reduce the risk of users being exposed to phishing, social engineering, or other malicious content by ensuring links are only accessible to signed-in members of your organization.

![A setting to require users to be logged in before following an external link from OpenProject](openproject_release_notes_17-2-0_external_links_log_in.png)

Read more about [capturing external links in OpenProject](../../system-admin-guide/system-settings/external-links/).

### Project Overview page improvements

OpenProject 17.2 enhances the Project Overview to provide clearer financial insights, easier inline editing, and improved accessibility. Together, these updates make the Overview page a more powerful and inclusive central hub for project information.

#### Updated Overview widget for Budgets

> Project, program, and portfolio managers can now see key financial indicators at a glance. New budget widget displays planned budget, actual costs, spent ratio, and remaining budget, along with visual breakdowns by cost type and recent monthly actuals. Data is automatically aggregated across subprojects where applicable, giving stakeholders a consolidated financial snapshot without leaving the Overview page.

Reviewer 3: ```suggestion
Project, program, and portfolio managers can now see key financial indicators at a glance. New budget widgets displays planned budget, actual costs, spent ratio, and remaining budget, along with visual breakdowns by cost type and recent monthly actuals. Data is automatically aggregated across subprojects where applicable, giving stakeholders a consolidated financial snapshot without leaving the Overview page.
```

> This widget helps teams better understand financial status and trends directly within their project context. Keep in mind that both the Budgets and Time & Costs modules need to be enabled for the widget to work. 

Reviewer 3: ```suggestion
These widgets helps teams better understand financial status and trends directly within their project context. Keep in mind that both the Budgets and Time & Costs modules need to be enabled for the widget to work. 
```

OpenProject 17.2 continues the transition to the Primer design system, further unifying the look and feel across the application.

#### Backlogs module update

The Backlogs module has been updated using Primer components. This resulted in a cleaner layout and more consistent interaction patterns, while still preserving familiar functionality such as drag-and-drop and version-based organization. Work packages can now be viewed in a split screen for improved context and efficiency.

![New UX/UI updates for the OpenProject Backlogs module](openproject_system_guide_new_backlog.png)

Read more about [Backlogs](../../user-guide/backlogs-scrum/).

#### Improvements in administration interface

Administrative interfaces for Custom Fields, Versions, and Groups have been further aligned with Primer. 

In particular, custom field forms are now consistently styled across all field types. Previously, the appearance varied depending on the type of custom field. This has been unified to provide a clearer and more predictable configuration experience for administrators.

## Important updates and breaking changes

**“Enable REST web service” renamed**

> The system setting previously labeled “Enable REST web service” is now called “REST API enabled”. This is a naming change only and does not affect functionality.

Reviewer: Factually wrong. Correct:

```suggestion
The system setting previously labeled “Enable REST web service” is now called “Enable API tokens”. This is a naming change only and does not affect functionality.
```

**“Status” boards renamed to “Kanban” boards**
To better reflect their purpose and common terminology, Status boards are now called Kanban boards. Existing boards and configurations remain unchanged.

**Improved OAuth token security for document collaboration**
OAuth tokens used for collaborative document editing (BlockNote ↔ Hocuspocus) now have shorter lifetimes and are automatically refreshed. This enhances security while keeping the editing experience unchanged.

> 

Reviewer: Something that was developed in the scope of the MCP Epic, but actually works independently of it _might_ deserve some attention here (I assume it does, because we even highlight renamings of config options here):

```markdown
**API tokens usable as Bearer token**
Newly generated API tokens can directly be used as Bearer tokens and do not need to be presented as basic auth credentials with the username `apikey`. This is intended to make usage of our APIs easier. The previously existing basic auth flow is still supported.
```

via: https://community.openproject.org/wp/71147

> 

Author: ```suggestion
**API tokens usable as Bearer token**
Newly generated API tokens can directly be used as Bearer tokens and do not need to be presented as basic auth credentials with the username `apikey`. This is intended to make usage of our APIs easier. The previously existing basic auth flow is still supported.
```

## Important feature changes

Take a look at our release video showing the most important features introduced in OpenProject 17.2.0:

![Release video of OpenProject 17.2]( UPDATE LINK) 

### MCP Server (Enterprise add-on)

[feature: mcp_server ]

> OpenProject 17.2 introduces the **MCP Server**, a new Enterprise add-on that lays the foundation for robust integrations between OpenProject and systems that use the Model Context Protocol (MCP), such as large language models or automation tools. This server exposes OpenProject’s APIv3 resources as MCP-compatible endpoints and enables secure, authenticated access for clients such as large language models or other MCP clients, opening the door to richer contextual interactions with your project data. 

Reviewer 2: I would:
* Add the abbreviation (LLMs) behind 'large language models' as this is what more people know. 
* Mention AI in this first paragraph. The integration between OP and an AI is where the value is in for the users.

Included in this release are administrative UI support for configuring the MCP Server, infrastructure and metadata endpoints, and integration of MCP authentication with OpenProject’s OAuth2 and API key mechanisms, including external OpenID Connect providers. An initial set of MCP tools and resources is provided to surface key entities (projects, work packages, users, etc.), and response formats can be adjusted based on your preferences. With session-cookie and bearer-token support, the MCP Server acts as a secure bridge between your OpenProject instance and external systems that operate via MCP. 

![Model context protocol settings in OpenProject administration](openproject_system_guide_new_mcp.png)

See the [**MCP Server documentation**](../../system-admin-guide/integrations/mcp-server/) for setup and examples.  

### Reusable meeting templates (Enterprise add-on)

[feature: meeting_templates ]

> Preparing meetings often involves recreating the same agenda structure again and again. With OpenProject 17.2, administrators can now define reusable meeting templates that provide a predefined agenda layout for their teams. These templates help ensure that important discussion points are consistently covered and that project meetings follow a clear, standardized structure.

Reviewer 2: To me it would help to even more highlight the idea behind this:

Even if the meeting is not necessarily a recurring one (but it might reoccur - just not regularly), it helps to propose a structure for meetings that can be reused. That way, best practices can find their way into meetings where a lot of people spend a lot of time.

Administrators can now require users to be logged in before following external links. When this setting is enabled, anyone who is not authenticated will be redirected to the login page before being allowed to continue to the external destination. This helps reduce the risk of users being exposed to phishing, social engineering, or other malicious content by ensuring links are only accessible to signed-in members of your organization.

![A setting to require users to be logged in before following an external link from OpenProject](openproject_release_notes_17-2-0_external_links_log_in.png)

Read more about [capturing external links in OpenProject](../../system-admin-guide/system-settings/external-links/).

### Project Overview page improvements

OpenProject 17.2 enhances the Project Overview to provide clearer financial insights, easier inline editing, and improved accessibility. Together, these updates make the Overview page a more powerful and inclusive central hub for project information.

#### Updated Overview widget for Budgets

> Project, program, and portfolio managers can now see key financial indicators at a glance. New budget widgets displays planned budget, actual costs, spent ratio, and remaining budget, along with visual breakdowns by cost type and recent monthly actuals. Data is automatically aggregated across subprojects where applicable, giving stakeholders a consolidated financial snapshot without leaving the Overview page.

Reviewer 2: "New budget widgets displays" - this should read "New budget widgets display"

> These widgets helps teams better understand financial status and trends directly within their project context. Keep in mind that both the Budgets and Time & Costs modules need to be enabled for the widget to work. 

Reviewer 2: "These widgets helps" -> "These widgets help"

> These widgets helps teams better understand financial status and trends directly within their project context. Keep in mind that both the Budgets and Time & Costs modules need to be enabled for the widget to work. 

Reviewer 2: " enabled for the widget to work". Is it one widget or multiple. I am confused.

![An example of project budget widget on a project home page in OpenProject](openproject_release_notes_17-2-0_budget_widget.png)

Read more about [budget widgets](../../user-guide/project-home/project-widgets/#budget-widget).

#### Editable project description and project status widgets on a Project view tab

> The project description and project status widgets on the Overview tab are now editable inline. Based on your feedback, we’ve streamlined the experience so authorized users can update content directly where they view it,  without switching to another tab.

Reviewer 2: "view it,  without" (extra white space)

![Overview tab of a project home page in OpenProject, showing the project description widget being edited](openproject_system_guide_project_home_overview_tab_edit.png)

#### Improves accessibility of Project Overview and Dashboard Widgets

We have significantly improved the accessibility of widgets on both the Project Overview and Project dashboard pages. Widgets are now fully operable via keyboard, provide clearer structural semantics for screen readers, and follow WCAG 2.1 AA guidelines for focus management, labeling, and navigation order.

These improvements ensure that project information and controls are accessible to all users, including those relying on assistive technologies.

### Comment fields for project attributes 

OpenProject 17.2 introduces optional comment fields for project attributes, giving portfolio and project managers additional context behind selected values. Administrators can now enable a dedicated comment field for individual project attributes. This allows users to document the reasoning, assumptions, or background information related to a specific attribute value directly where it is maintained.

> Comments are displayed and edited alongside the respective attribute on the Project overview page and follow the same permission logic as the attribute itself. Changes are tracked in the project activity, included in exports, and available via the API. By adding structured context to project metadata, this enhancement improves transparency and supports better governance and decision-making across projects and teams.

Reviewer 2: I like this paragraph 👍

![Setting to add a comment text field to a project atttribute in OpenProject administration](openproject_release_notes_17-2-0_project_attributes_comment.png)

Read more about [project attributes in OpenProject](../../user-guide/project-home/project-attributes/).

### PDF export improvements

> OpenProject 17.2 enhances PDF exports to provide more complete and reliable reporting.

Reviewer 2: "more complete" sounds weird. Maybe "exhaustive" or "thorough"?

- Feature: Allow requiring to be logged in for external links [[#71624](https://community.openproject.org/wp/71624)]
- Feature: Primerize versions project settings [[#71641](https://community.openproject.org/wp/71641)]
- Feature: Primerize groups administration [[#71642](https://community.openproject.org/wp/71642)]
- Feature: Rename "Enable REST web service" setting [[#71886](https://community.openproject.org/wp/71886)]
- Feature: Reduce page size of MCP responses [[#71977](https://community.openproject.org/wp/71977)]
- Feature: Allow to configure MCP tool response volume [[#71978](https://community.openproject.org/wp/71978)]
- Feature: Allow authentication to MCP endpoint via session cookie [[#72253](https://community.openproject.org/wp/72253)]
- Feature: Enable Column Sorting on Versions Overview [[#72354](https://community.openproject.org/wp/72354)]
- Feature: MCP Server as a bridge between OpenProject and LLMs [[#62781](https://community.openproject.org/wp/62781)]

<!-- END AUTOMATED SECTION -->
<!-- Warning: Anything above this line will be automatically removed by the release script -->

## Contributions

> A very special thank you goes to our sponsors for this release. UPDATE

Reviewer 3: The "UPDATE" probably needs to be removed?

Your support, alongside the efforts of our amazing Community, helps drive these innovations.

Also a big thanks to our Community members for reporting bugs and helping us identify and provide fixes.
Special thanks for reporting and finding bugs go to Holger Schantin, Stefan Weiberg, Jure Uršič, Natalie Stettner, Romain Besson.

Last but not least, we are very grateful for our very engaged translation contributors on Crowdin, who translated quite a few OpenProject strings! This release we would like to particularly thank the following users:

> UPDATE

Reviewer 3: Same as above?
